### ISSUE TITLE
Unable to allow unused_self after enabling pedantic

### ISSUE THREAD

**Issue by martinthomson** [rust-lang/rust-clippy#5053]
[This code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=45b7784b35beabe68fbd28b12a58285e):

```rust
#![deny(warnings)]
#![warn(clippy::pedantic)]

struct X {}
impl X {
    #[allow(clippy::unused_self, clippy::cast_possible_truncation)]
    fn test(&self, a: u32) {
        println!("testing {}", a as u16);
    }
}
pub fn main() {
    let x = X {};
    x.test(1);
}
```

This correctly suppresses the `clippy::cast_possible_truncation` warning, but fails to suppress the `clippy::unused_self`.

The rust playground is currently at 0.0.212 (2019-12-29 c807fbc).
The same error is produced by clippy 0.0.212 (c8e3cfb 2019-10-28).
Older versions I have access to, such as `clippy 0.0.212 (4e7e71b 2019-10-11)` don't have this lint, so it's hard to pin down.

**Comment by taiki-e** [comment]
A workaround for this is to use `#[allow(clippy::unused_self)]` on `impl` block: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3e9ed16b05b8b6111754f37d6655cac0)
